### ISSUE TITLE
Evaluate suitability of NeXML data extracting functions

### ISSUE THREAD

**Issue by johnbradley** [phenoscape/rphenoscape#172]
Determine if the NeXML data extracting functions would better as additions to [RNeXML](https://github.com/ropensci/RNeXML).

There are several data extracting functions that process NeXML objects :
- [pk_get_ontotrace(nex)](https://github.com/phenoscape/rphenoscape/blob/788230e8222a02e7536d9106ec0d211b60d9f8f8/R/pk_ontotrace.R#L23-L28) - this function just directly calls `RNeXML::get_characters`
- [pk_get_ontotrace_meta(nex)](https://github.com/phenoscape/rphenoscape/blob/788230e8222a02e7536d9106ec0d211b60d9f8f8/R/pk_ontotrace.R#L34-L60)
- [pk_get_study(nexmls)](https://github.com/phenoscape/rphenoscape/blob/b367c4416af4cc6fa45a00765ce45a8d412b5c22/R/pk_study.R#L133-L145) which is just __lapply(nexmls, [pk_get_study_by_one](https://github.com/phenoscape/rphenoscape/blob/b367c4416af4cc6fa45a00765ce45a8d412b5c22/R/pk_study.R#L174-L206))__
- [pk_get_study_meta(nexmls)](https://github.com/phenoscape/rphenoscape/blob/b367c4416af4cc6fa45a00765ce45a8d412b5c22/R/pk_study.R#L158-L170) which is just __lapply(nexmls, [pk_get_study_meta_by_one](https://github.com/phenoscape/rphenoscape/blob/b367c4416af4cc6fa45a00765ce45a8d412b5c22/R/pk_study.R#L209-L232))__

Determine if the functions processing a list of NeXML objects (pk_get_study*) can be dropped in favor of exporting the internal functions they call.

**Comment by johnbradley** [comment]
For now we plan on leaving these functions in rphenoscape.

Here is what I am thinking for renaming these functions:
- `pk_get_ontotrace(nex)` renamed to `get_ontotrace_char_matrix(nex)`
- `pk_get_ontotrace_meta(nex)` renamed to `get_ontotrace_meta(nex)`
- `pk_get_study(nexmls)` renamed to `get_study_char_matrix(nexmls)`
- `pk_get_study_meta(nexmls)` renamed to `get_study_meta(nexmls)`

Thoughts @hlapp ?

**Comment by johnbradley** [comment]
How about creating a function that supports both cases like so?
```
get_char_matrix <- function(nex, otus_id = TRUE, translate_symbols=FALSE) {...}
```

**Comment by hlapp** [comment]
I think that's a good way to think about it. `0`, `1`, `2` etc are symbols that would be used in a traditional character state matrix format that most analysis programs (and R packages for comparative analysis)  will expect (for categorical character states). (Note that technically, there is no such thing as a "range" of such numbers; there are no implied numeric semantics to `1` vs `0` or `3` other than that they signal distinct character states (much like nucleotide bases in genetic data).

Instead of `translate_symbols`, I'd suggest something like `states_as_symbols`, which should probably default to TRUE (because most analysis functions will expect symbols, not labels).

**Comment by hlapp** [comment]
> To me it seems like some meaning has been lost here with "1 and 0" becoming "".

The problem there is that the state for this taxon is polymorphic. I suspect the code can't handle that when using labels (it would have to say `present and absent`, for example). This sounds more like a bug.

**Comment by johnbradley** [comment]
Fixed by #228
